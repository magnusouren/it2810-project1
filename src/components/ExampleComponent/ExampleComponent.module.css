.heading {
  font-size: 2rem;
  font-weight: 700;
  margin-bottom: 1rem;
  text-align: center;
}
