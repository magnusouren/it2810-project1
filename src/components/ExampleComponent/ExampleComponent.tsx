import React, { FC } from 'react';

import styles from './ExampleComponent.module.css';

interface AccordionProps {
  title: string;
  children: React.ReactNode;
}

export const ExampleComponent: FC<AccordionProps> = ({ title, children }) => {
  return (
    <div>
      <h3 className={styles.heading}>{title}</h3>
      <div>{children}</div>
    </div>
  );
};
