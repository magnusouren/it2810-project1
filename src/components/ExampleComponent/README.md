# ExampleComponent

Dummy component for eksempelbruk.
