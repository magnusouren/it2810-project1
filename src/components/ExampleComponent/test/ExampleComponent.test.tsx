import { render, screen } from '@testing-library/react';
import { describe, it } from 'vitest';

import { ExampleComponent } from '../ExampleComponent';

// Les mer: https://eternaldev.com/blog/testing-a-react-application-with-vitest

describe('Accordion test', () => {
  it('Should match snapshot', () => {
    const { container } = render(
      <ExampleComponent title='Testing'>
        <h4>Content</h4>
      </ExampleComponent>,
    );

    expect(container).toMatchSnapshot();
  });

  it('Should show title', () => {
    render(
      <ExampleComponent title='Testing'>
        <h4>Content</h4>
      </ExampleComponent>,
    );

    expect(screen.getByText(/Testing/i)).toBeDefined();
    expect(screen.getByRole('heading', { name: /Testing/i })).toBeDefined();
  });
});
