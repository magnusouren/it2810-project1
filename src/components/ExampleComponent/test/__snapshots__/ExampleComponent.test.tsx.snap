// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion test > Should match snapshot 1`] = `
<div>
  <div>
    <h3
      class="_heading_d988dd"
    >
      Testing
    </h3>
    <div>
      <h4>
        Content
      </h4>
    </div>
  </div>
</div>
`;
